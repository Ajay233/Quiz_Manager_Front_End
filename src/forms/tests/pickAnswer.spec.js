import React from 'react'
import { Provider } from 'react-redux'
import PickAnswer from '../pickAnswer'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import mockAxios from 'jest-mock-axios'
import history from '../../history'

const mockStore = configureStore({})

jest.mock('../../history')

describe("pickAnswer form", () => {

  let store;
  let answers;
  let onSubmit;
  let exitQuiz;

  beforeEach(() => {
    store = mockStore({
      answers
    })

    answers = [{ id: 1, description: "test" }, { id: 2, description: "test2" }]
    onSubmit = jest.fn()
    exitQuiz = jest.fn()
  })

  it("should call the passed in submit method when the for is submitted", () => {
    const wrapper = mount(
      <Provider store={store}>
        <PickAnswer
          title={"Answers"}
          answers={answers}
          onSubmit={onSubmit}
          numberOfQuestions={2}
          currentQuestionNumber={1}
          showResults={false}
          exit={exitQuiz}
        />
      </Provider>
    )

    wrapper.find('form').simulate('submit')
    expect(onSubmit).toHaveBeenCalledTimes(1)
    // expect(reset).toHaveBeenCalledTimes(1)
  })

  it("should call the passed in quit method when the quit button is clicked", () => {
    const wrapper = mount(
      <Provider store={store}>
        <PickAnswer
          title={"Answers"}
          answers={answers}
          onSubmit={onSubmit}
          numberOfQuestions={2}
          currentQuestionNumber={1}
          showResults={false}
          exit={exitQuiz}
        />
      </Provider>
    )

    wrapper.find('button').at(1).simulate('click')
    expect(history.push).toHaveBeenCalledTimes(1)
    expect(exitQuiz).toHaveBeenCalledTimes(1)
  })

  it("should match the snapshot, button title should be: 'Next Question'", () => {
    const component = renderer.create(
      <Provider store={store}>
        <PickAnswer
          title={"Answers"}
          answers={answers}
          onSubmit={onSubmit}
          numberOfQuestions={2}
          currentQuestionNumber={1}
          showResults={false}
          exit={exitQuiz}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it("should match the snapshot, button title should be: 'Finish'", () => {
    const component = renderer.create(
      <Provider store={store}>
        <PickAnswer
          title={"Answers"}
          answers={answers}
          onSubmit={onSubmit}
          numberOfQuestions={2}
          currentQuestionNumber={2}
          showResults={false}
          exit={exitQuiz}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })
})
