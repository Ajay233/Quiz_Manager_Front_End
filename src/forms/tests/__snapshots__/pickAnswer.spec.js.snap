// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pickAnswer form should match the snapshot, button title should be: 'Finish' 1`] = `
<div
  id="pickAnswerForm"
>
  <form
    onSubmit={[Function]}
  >
    <div
      id="answerContainer"
    >
      <div
        className="title-medium-left-alt"
      >
        Answers
      </div>
      <div>
        <label>
          <input
            checked={false}
            className="radioInput"
            name="answer"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="radio"
            value="test"
          />
          test
        </label>
      </div>
      <div>
        <label>
          <input
            checked={false}
            className="radioInput"
            name="answer"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="radio"
            value="test2"
          />
          test2
        </label>
      </div>
    </div>
    <img
      alt=""
      src="reply.png"
    />
    <div
      className="buttonContainer"
    >
      <button
        className="submit quizButton-left"
        data-testid="next-button"
      >
        Finish
      </button>
      <button
        className="submit quizButton-right"
        data-testid="quit-button"
        onClick={[Function]}
      >
        Quit
      </button>
    </div>
  </form>
</div>
`;

exports[`pickAnswer form should match the snapshot, button title should be: 'Next Question' 1`] = `
<div
  id="pickAnswerForm"
>
  <form
    onSubmit={[Function]}
  >
    <div
      id="answerContainer"
    >
      <div
        className="title-medium-left-alt"
      >
        Answers
      </div>
      <div>
        <label>
          <input
            checked={false}
            className="radioInput"
            name="answer"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="radio"
            value="test"
          />
          test
        </label>
      </div>
      <div>
        <label>
          <input
            checked={false}
            className="radioInput"
            name="answer"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="radio"
            value="test2"
          />
          test2
        </label>
      </div>
    </div>
    <img
      alt=""
      src="reply.png"
    />
    <div
      className="buttonContainer"
    >
      <button
        className="submit quizButton-left"
        data-testid="next-button"
      >
        Next Question
      </button>
      <button
        className="submit quizButton-right"
        data-testid="quit-button"
        onClick={[Function]}
      >
        Quit
      </button>
    </div>
  </form>
</div>
`;
